### PR TITLE
Feature/pxblue 2537 react status list

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes PXBLUE-2537

#### Changes proposed in this Pull Request:
- Changed the random display status list
- Added accordions for each section
- Modified list items to have icons, timestamp, title, subtitle, tags, chevron
- Added mobile view changes

#### Screenshots:
**Web**
**Before**
PXBLUE-2537-BEFORE-WEB

**After**
Screenshot 2021-10-11 at 8 42 05 PM

**Mobile**
**Before**
PXBLUE-2537-BEFORE-MBL
**After**
Screenshot 2021-10-11 at 8 45 14 PM

#### To Test:
- http://localhost:3000/status-list

### CODE REVIEW

**Comment by bkarambe** [comment]
> Just eyeballing it visually:
> 
> 1. Can you add missing dates here:
>    image
> 2. Can you center the page content on wider viewports?
>    image
> 3. Can you add in these icons to the app bar?
>    image

@emclaug2  Updated the pr with suggested  changes.

File: src/pages/lists/status-list/index.tsx [hunk 1]
```diff
@@ -1,78 +1,267 @@
...
+                style={{
+                    display: 'flex',
+                    margin: '4px 0px 4px 32px',
+                }}
+            >
+                <ListItemTag label={'assigned'} backgroundColor={colors.blue[500]} />
+                <ListItemTag
+                    label={'active'}
+                    backgroundColor={colors.red[500]}
+                    style={{ marginLeft: '16px', marginRight: '32px' }}
```
Comment (ektasawant) [inline src/pages/lists/status-list/index.tsx:168]: Can we remove inline style?

File: src/pages/lists/status-list/index.tsx [hunk 2]
```diff
@@ -1,78 +1,267 @@
...
+        <Chevron color={'inherit'} role={'button'} />
+    </div>
+);
+
+const getInfoComponent = (tag: boolean, isMobile: boolean): string | Array<string | JSX.Element> | undefined => {
+    if (tag && isMobile) {
+        return [
+            <div
+                key="info"
+                style={{
```
[the remark this answers is not in the document]
Comment (bkarambe) [reply src/pages/lists/status-list/index.tsx]: Updated the changes.

**Comment by bkarambe** [comment]
> ^ the screenshot above has the chevron pointing in the opposite direction.
> 
> When something is collapsed, the chevron points down; when something is expanded, the chevron points up.

Done. Pushed the changes. The only change pending is removing the line above accordion, when collapsed.

File: src/pages/lists/status-list/index.tsx [hunk 3]
```diff
@@ -37,28 +38,51 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+        [theme.breakpoints.down('sm')]: {
+            margin: `0 0 ${theme.spacing(3)}px 0`,
+            paddingTop: 0,
+        },
+    },
     accordionRoot: {
-        width: '768px',
+        marginBottom: theme.spacing(2),
         boxShadow:
             '0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px rgba(0, 0, 0, 0.14), 0px 1px 3px rgba(0, 0, 0, 0.12)',
```
Comment (huayunh) [inline src/pages/lists/status-list/index.tsx:59]: why is this box shadow hard coded

**Comment by jeffvg** [comment]
looks good. I did notice while in FireFox "emulator" view when clicking the accordions it kicks up this dep warning. not sure if it will be an issue in future

image

**Comment by emclaug2** [comment]
> I noticed this distance on the drawer menu isn't correct, there should be a 72px distance on the secondary options

Since this is outside the scope for this story I've made a small issue for this.  Here #105

File: src/pages/lists/status-list/index.tsx [hunk 4]
```diff
@@ -1,78 +1,371 @@
...
     },
     toolbarGutters: {
-        padding: '0 16px',
+        padding: `0 ${theme.spacing(2)}px`,
+        display: 'flex',
+        justifyContent: 'space-between',
+    },
+    listItemTextWithoutIcon: {
+        marginLeft: -theme.spacing(7),
+    },
```
Comment (daileytj) [inline src/pages/lists/status-list/index.tsx:42]: Why the negative margin here? There is a `hidePadding` prop on the `InfoListItem` component you can use to handle this.

File: src/pages/lists/status-list/index.tsx [hunk 5]
```diff
@@ -1,78 +1,371 @@
...
+    toolbarRightContent: {
+        display: 'flex',
+        flexDirection: 'row',
+    },
+    listItemTitle: {
+        display: 'flex',
+        alignItems: 'center',
+        color: colors.black[500],
+        fontWeight: 400,
+        marginLeft: theme.spacing(4),
```
Comment (daileytj) [inline src/pages/lists/status-list/index.tsx:99]: Shouldn't need this as the styles for the title, subtitle, and info spacing should already be taken care of in the underlying component. You can probably just add 16px margin to the left content and remove all the leftMargin styles from title, subtitle, and info.

File: src/pages/lists/status-list/index.tsx [hunk 6]
```diff
@@ -1,78 +1,371 @@
...
+    },
+    listItemTitleWithoutTimeStamp: {
+        marginLeft: 'auto',
+    },
+    station: {
+        fontSize: 14,
+        fontWeight: 400,
+        marginLeft: theme.spacing(4),
```
Comment (daileytj) [inline src/pages/lists/status-list/index.tsx:107]: see the previous comment about spacing for title, subtitle, and info.

File: src/pages/lists/status-list/index.tsx [hunk 7]
```diff
@@ -1,78 +1,371 @@
...
+        fontWeight: 400,
+    },
+    leftComponentRoot: {
+        display: 'flex',
+        flexDirection: 'column',
+        fontSize: '12px',
+    },
+    leftComponentRootWithoutIcon: {
+        marginLeft: -theme.spacing(7),
+    },
```
Comment (daileytj) [inline src/pages/lists/status-list/index.tsx:125]: use `hidePadding` prop on `InfoListItem`

File: src/pages/lists/status-list/index.tsx [hunk 8]
```diff
@@ -1,78 +1,371 @@
...
+    timePeriod: {
+        marginLeft: '4px',
+    },
+    rightComponentRoot: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    assignedTag: {
+        marginLeft: theme.spacing(4),
```
Comment (daileytj) [inline src/pages/lists/status-list/index.tsx:143]: shouldn't need this... see previous marginLeft comments

File: src/pages/lists/status-list/index.tsx [hunk 9]
```diff
@@ -1,78 +1,371 @@
...
+        marginTop: 4,
+    },
+    activeTag: {
+        marginLeft: 0,
```
Comment (daileytj) [inline src/pages/lists/status-list/index.tsx:147]: shouldn't need this... see previous marginLeft comments

File: src/pages/lists/status-list/index.tsx [hunk 10]
```diff
@@ -1,78 +1,371 @@
...
+    },
+    accordionContainer: {
+        maxWidth: 768,
+        paddingTop: theme.spacing(3),
+        margin: '0 auto',
+
+        [theme.breakpoints.down('sm')]: {
+            margin: `0 0 ${theme.spacing(3)}px 0`,
+            paddingTop: 0,
+        },
```
Comment (daileytj) [inline src/pages/lists/status-list/index.tsx:54]: For some reason this is breaking down at 960 while everything else breaks down at 600 leaving this strange behavior between 600-960px.
Screen Shot 2021-10-14 at 2 12 28 PM

File: src/pages/lists/status-list/index.tsx [hunk 11]
```diff
@@ -1,78 +1,371 @@
...
+        [theme.breakpoints.down('sm')]: {
+            maxWidth: '100%',
+            boxShadow: 'none',
+            borderRadius: 0,
+            '&:before': {
+                display: 'none',
+            },
+            '& .MuiAccordionSummary-root': {
+                height: '48px',
```
Comment (huayunh) [inline src/pages/lists/status-list/index.tsx:73]: Always avoid absolute pixel values like this for spacing

File: src/pages/lists/status-list/index.tsx (hunk 5 above)
Comment (huayunh) [reply src/pages/lists/status-list/index.tsx]: This ^

Suraj, feel free to take some time to explore around the info list item component and get familiar with our API.

File: src/pages/lists/status-list/index.tsx [hunk 12]
```diff
@@ -1,78 +1,371 @@
...
+    activeTag: {
+        marginLeft: 0,
+        marginTop: 4,
+    },
+    listItemTag: {
+        marginLeft: theme.spacing(2),
+        marginRight: theme.spacing(4),
+    },
+    rightComponentChevron: {
+        color: colors.gray[500],
```
Comment (huayunh) [inline src/pages/lists/status-list/index.tsx:139]: shouldn't need this either

File: src/pages/lists/status-list/index.tsx [hunk 13]
```diff
@@ -1,78 +1,371 @@
...
+);
+
+const getRightComponent = (isMobile: boolean, tag: boolean, classes: Record<string, any>): ReactNode => (
+    <div className={classes.rightComponentRoot}>
+        {tag && !isMobile && (
+            <>
+                <ListItemTag label={'assigned'} backgroundColor={colors.blue[500]} />
+                <ListItemTag
+                    label={'active'}
+                    backgroundColor={colors.red[500]}
```
Comment (huayunh) [inline src/pages/lists/status-list/index.tsx:202]: again, avoid using @pxblue/colors directly here. Usually you can pull things out of theme.palette... except that orange color on this page.

You can take a peek at MUI's default theme here: https://mui.com/customization/default-theme/

File: src/pages/lists/status-list/index.tsx [hunk 14]
```diff
@@ -89,23 +356,81 @@ export const StatusList = (): JSX.Element => {
...
+                    <div className={classes.toolbarRightContent}>
+                        <IconButton color={'inherit'}>
+                            <HelpIcon />
+                        </IconButton>
+                        <IconButton color={'inherit'}>
+                            <Badge color="error" badgeContent={88}>
+                                <NotificationIcon />
+                            </Badge>
+                        </IconButton>
+                        <IconButton color={'inherit'}>
```
Comment (huayunh) [inline src/pages/lists/status-list/index.tsx:371]: `edge={'end'}` can help

**Review by huayunh — COMMENTED** [review]
>  Do we want to add spaces in between the + signs in "With Time Stamps, with Title+Subtitle+Info"? and do we want to add "load more events" to the bottom of any of the groups?

Neither happens to be the focus of this pattern, which focuses on the variation of list item structure. I'd say these are not necessary

**Comment by huayunh** [comment]
The mobile breakpoint issue seems to be still there

File: src/pages/lists/status-list/index.tsx (hunk 11 above)
Comment (bkarambe) [reply src/pages/lists/status-list/index.tsx]: Done.

File: src/pages/lists/status-list/index.tsx (hunk 14 above)
Comment (bkarambe) [reply src/pages/lists/status-list/index.tsx]: Done

File: src/pages/lists/status-list/index.tsx (hunk 7 above)
Comment (bkarambe) [reply src/pages/lists/status-list/index.tsx]: Done.

File: src/pages/lists/status-list/index.tsx (hunk 6 above)
Comment (bkarambe) [reply src/pages/lists/status-list/index.tsx]: Done

File: src/pages/lists/status-list/index.tsx (hunk 5 above)
Comment (bkarambe) [reply src/pages/lists/status-list/index.tsx]: Done.

File: src/pages/lists/status-list/index.tsx (hunk 4 above)
Comment (bkarambe) [reply src/pages/lists/status-list/index.tsx]: Done

File: src/pages/lists/status-list/index.tsx (hunk 12 above)
Comment (huayunh) [reply src/pages/lists/status-list/index.tsx:135]: Can you log a bug in our React component library repo for this? By default, the chevron (and any non-text `rightComponent`) should be `theme.palette.text.secondary`.

File: src/pages/lists/status-list/index.tsx [hunk 15]
```diff
@@ -89,23 +247,177 @@ export const StatusList = (): JSX.Element => {
...
+                                icon={<NotificationsActiveIcon />}
+                                iconColor={theme.palette.common.white[50]}
+                                statusColor={colors.red[500]}
+                                leftComponent={getLeftComponent('8:21', 'AM', '11/23/21', classes)}
+                                rightComponent={getRightComponent(isMobile, classes, true)}
+                                divider={'partial'}
+                                avatar
+                                chevron
+                            />
+                            <InfoListItem
```
Comment (huayunh) [inline src/pages/lists/status-list/index.tsx:294]: You would want the `iconAlign` prop of all `<InfoListItem>` set to `'center'`. Read more about the rule here: https://pxblue.github.io/patterns/lists#lists-with-status

**Review by huayunh — COMMENTED** [review]
Please double check. The breakpoint change should happen at 600px wide.

https://user-images.githubusercontent.com/8997218/137805099-09c8c3ec-fb94-4cfe-8bb2-90e457ba898f.mov

File: src/pages/lists/status-list/index.tsx (hunk 15 above)
Comment (bkarambe) [reply src/pages/lists/status-list/index.tsx:294]: Done.

**Pull request merged**